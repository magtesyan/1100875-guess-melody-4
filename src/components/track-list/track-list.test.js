import React from "react";
import TrackList from "./track-list.jsx";
import renderer from "react-test-renderer";

const userAnswers = [false, true, false, false];
const answer = {
  id: `12345`,
  src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
  genre: `blues`,
};
const index = 1;

it(`TrackList is rendered correctly`, () => {
  const handleAnswerSubmit = jest.fn();
  const tree = renderer
    .create(<TrackList
      src = {answer.src}
      index = {index}
      key = {answer.id}
      userAnswers = {userAnswers}
      handleAnswerSubmit = {handleAnswerSubmit}
      renderPlayer={() => {}}
    />
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
