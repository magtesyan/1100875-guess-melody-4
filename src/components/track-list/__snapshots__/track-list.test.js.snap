// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrackList is rendered correctly 1`] = `
<div
  className="track"
>
  <button
    className="track__button track__button--play"
    disabled={true}
    onClick={[Function]}
    type="button"
  />
  <div
    className="track__status"
  >
    <audio />
  </div>
  <div
    className="game__answer"
  >
    <input
      checked={true}
      className="game__input visually-hidden"
      id="answer-1"
      name="answer"
      onChange={[Function]}
      type="checkbox"
      value="answer-1"
    />
    <label
      className="game__check"
      htmlFor="answer-1"
    >
      Отметить
    </label>
  </div>
</div>
`;
